src/drivers/gitlab.js

```diff
@@ -53,14 +53,21 @@ class Gitlab {
                 .version
             )
               return path;
-          } catch (error) {}
+          } catch (error) {
+            return error;
+          }
         })
     );
 
-    this._detected_base = possibleBases.find(Boolean);
-    if (!this._detected_base) throw new Error('GitLab API not found');
+    this.detectedBase = possibleBases.find(
+      (base) => base.constructor !== Error
+    );
+    if (!this.detectedBase) {
+      if (possibleBases.length) throw possibleBases[0];
+      else throw new Error('Invalid repository address');
```

Comment:
Do this ever happen? If so the else is not needed.

Reply:
The `else` condition would only happen when `pathname` contains an invalid value. Maybe a bit defensive, but grant us a meaningful thrown error in every case.

Reply:
```else``` not needed because if happens the first throw if not throw so else is not needed.
What m not sure is that if ever happen. X requests returns error always except if they succeeded (ideally only once) so I can see a [X-1, X] range Errors at least? Assuming that fetch will return Error

Reply:
> `else` not needed because if happens the first throw if not throw so else is not needed.
👍🏼  🦅  👁️ 

> _What m not sure is that if ever happen. X requests returns error always except if they succeeded (ideally only once) so I can see a [X-1, X] range Errors at least? Assuming that fetch will return Error_

This would only happen when no requests are made.

Reply:
But we assume that fetch will do a request, right? 🤔

Reply:
For `this.repo === 'https://gitlab.com'` we're doing zero requests. Of course, this is a blatantly invalid addess, but we must signal it somehow for users to notice in case they're feeding CML a malformed address due to an error.

Reply:
thats reminds me that we have a special repo 'cml' for cml-publish, might this introduce a break?

Reply:
If `this.repo === 'cml'` and tests reach this section of the code, you'll get the `Invalid repository address` error. Nevetheless, this edge case also raised an error before, so I guess that we're fine; tests are passing.